packages/plugin-ext/src/main/browser/debug/plugin-debug-service.ts

```diff
@@ -104,52 +99,72 @@ export class PluginDebugService implements DebugService, PluginDebugAdapterContr
     }
 
     async provideDebugConfigurations(debugType: keyof DebugRequestTypes, workspaceFolderUri: string | undefined): Promise<theia.DebugConfiguration[]> {
-        const contributor = this.contributors.get(debugType);
-        if (contributor) {
-            return contributor.provideDebugConfigurations && contributor.provideDebugConfigurations(workspaceFolderUri) || [];
-        } else {
+        const pluginProviders =
+            Array.from(this.configurationProviders.values()).
+                filter(
+                    p => p.triggerKind === DebugConfigurationProviderTriggerKind.Initial &&
+                    ( p.type === debugType || p.type === '*') &&
+                    !!p.provideDebugConfigurations);
+
+        if (pluginProviders.length === 0) {
             return this.delegated.provideDebugConfigurations(debugType, workspaceFolderUri);
         }
-    }
-
-    async provideDynamicDebugConfigurations(): Promise<{ type: string, configurations: DebugConfiguration[] }[]> {
-        const result: Promise<{ type: string, configurations: theia.DebugConfiguration[] }>[] = [];
 
-        for (const [type, contributor] of this.contributors.entries()) {
-            const typeConfigurations = this.resolveDynamicConfigurationsForType(type, contributor);
-            result.push(typeConfigurations);
+        let results: DebugConfiguration[] = [];
+        for (const provider of pluginProviders) {
+            const result = await provider.provideDebugConfigurations(workspaceFolderUri);
+            if (result) {
+                results = results.concat(result);
```

Comment:
Minor: I think `results.push(...result)` would be a little better than creating a new array each time.